2019/intcode.d

```diff
@@ -18,19 +18,37 @@ struct IntCode {
     bool p1imm = (opcode / 100) % 10 == 1;
     bool p2imm = (opcode / 1000) % 10 == 1;
 
+    int v(int n, bool imm) {
+      return imm ? memory[n] : memory[memory[n]];
+    }
+
     instr.op = to!Opcode(opcode % 100);
     final switch (instr.op) {
+      case Opcode.JT:
+        instr.ld1 = v(PC+1, p1imm);
```

Comment:
Compiler Explorer shows that [LDC 1.18.0 with -O3](https://godbolt.org/z/W4OKvi) optimises these function calls away.